File: C++/rps.cpp
#include <iostream>
#include <ctime>

char getUserChoice();
char getComputerChoice();
void showChoice(char choice);
void chooseWinner(char player, char computer);

int main() {

    char player;
    char computer;

    player = getUserChoice();
    std::cout << "You chose: ";
    showChoice(player);
    computer = getComputerChoice();
    std::cout << "Computer chose: ";
    showChoice(computer);

    chooseWinner(player, computer);

    return 0;
}

char getUserChoice() {

    char player;
    do {
        std::cout << "Welcome to Rock-Paper-Scissors!" << std::endl;
        std::cout << "Please choose: r for Rock, p for Paper, s for Scissors" << std::endl;
        std::cout << "Enter your choice (r/p/s): ";
        std::cin >> player;
    } while (player != 'r' && player != 'p' && player != 's');

    return player;
}

char getComputerChoice() {
    srand(time(0));
    int num = rand() % 3;
    if (num == 0) return 'r';
    else if (num == 1) return 'p';
    else return 's';
}

void showChoice(char choice) {
 
    switch(choice) {
        case 'r':
            std::cout << "Rock." << std::endl;
            break;
        case 'p':
            std::cout << "Paper." << std::endl;
            break;
        case 's':
            std::cout << "Scissors." << std::endl;
            break;
        default:
            std::cout << "Invalid choice." << std::endl;
    }
}

void chooseWinner(char player, char computer) {

    if (player == computer) {
        std::cout << "It's a tie!" << std::endl;
    } else if ((player == 'r' && computer == 's') ||
               (player == 'p' && computer == 'r') ||
               (player == 's' && computer == 'p')) {
        std::cout << "You win!" << std::endl;
    } else {
        std::cout << "Computer wins!" << std::endl;
    }
}

File: C#/rps.cs
using System;

namespace MyApp;

class Program3 {
    public static void rps() {
        
        Random random = new Random();
        string[] choices = { "rock", "paper", "scissors" };
        bool playAgain = true;

        while (playAgain) {
            Console.WriteLine("------------------------------------------");
            Console.WriteLine("Welcome to the Rock, Paper, Scissors Game!");
            Console.WriteLine("------------------------------------------");

            Console.WriteLine("Enter rock, paper or scissors:");
            string userChoice = Console.ReadLine().ToLower();

            if (Array.Exists(choices, element => element == userChoice)) {
                string computerChoice = choices[random.Next(choices.Length)];
                Console.WriteLine($"Computer chose: {computerChoice}");

                if (userChoice == computerChoice) {
                    Console.WriteLine("It's a tie!");
                } else if ((userChoice == "rock" && computerChoice == "scissors") ||
                           (userChoice == "paper" && computerChoice == "rock") ||
                           (userChoice == "scissors" && computerChoice == "paper")) {
                    Console.WriteLine("You win!");
                } else {
                    Console.WriteLine("You lose!");
                }
            } else {
                Console.WriteLine("Invalid choice. Please try again.");
            }

            Console.WriteLine("Do you want to play again? (yes/no)");
            string response = Console.ReadLine().ToLower();
            playAgain = response == "yes";
        }
    }
}
